extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/music/GoogleMusicExporter.java

```diff
@@ -0,0 +1,281 @@
+    } else if (paginationDataPresent && paginationToken.getToken().startsWith(TRACK_TOKEN_PREFIX)) {
+      // TODO: export tracks
+      return new ExportResult<>(ResultType.END, null, null);
+    } else if (paginationDataPresent
+        && paginationToken.getToken().startsWith(RELEASE_TOKEN_PREFIX)) {
+      // TODO: export releases
+      return new ExportResult<>(ResultType.END, null, null);
+    } else {
+      // There is nothing to export.
+      return new ExportResult<>(ResultType.END, null, null);
+    }
+  }
+
+  @VisibleForTesting
+  ExportResult<MusicContainerResource> exportPlaylists(
+      TokensAndUrlAuthData authData, Optional<PaginationData> paginationData, UUID jobId)
+      throws IOException, InvalidTokenException, PermissionDeniedException {
+    Optional<String> paginationToken = Optional.empty();
+    String prefixToken = "";
+    if (paginationData.isPresent()) {
+      String token = ((StringPaginationToken) paginationData.get()).getToken();
+      Preconditions.checkArgument(
+          token.startsWith(PLAYLIST_TOKEN_PREFIX), "Invalid pagination token %s", token);
+      prefixToken = token.substring(0, getTokenPrefixLength(token));
+      if (getTokenPrefixLength(token) < token.length()) {
+        paginationToken = Optional.of(token.substring(getTokenPrefixLength(token)));
+      }
+    }
+
+    PlaylistListResponse playlistListResponse =
+        getOrCreateMusicInterface(authData).listPlaylists(paginationToken);
+
+    PaginationData nextPageData;
+    String token = playlistListResponse.getNextPageToken();
+    List<MusicPlaylist> playlists = new ArrayList<>();
+    GooglePlaylist[] googlePlaylists = playlistListResponse.getPlaylists();
+
+    if (Strings.isNullOrEmpty(token)) {
+      nextPageData =
+          new StringPaginationToken(prefixToken.substring(PLAYLIST_TOKEN_PREFIX.length()));
+    } else {
+      nextPageData = new StringPaginationToken(prefixToken + token);
+    }
+    ContinuationData continuationData = new ContinuationData(nextPageData);
+
+    if (googlePlaylists != null && googlePlaylists.length > 0) {
+      for (GooglePlaylist googlePlaylist : googlePlaylists) {
+        MusicPlaylist musicPlaylist =
+            new MusicPlaylist(
+                googlePlaylist.getName().substring(GOOGLE_PLAYLIST_NAME_PREFIX.length()),
+                googlePlaylist.getTitle(),
+                googlePlaylist.getDescription(),
+                Instant.ofEpochMilli(googlePlaylist.getCreateTime()),
+                Instant.ofEpochMilli(googlePlaylist.getUpdateTime()));
+        playlists.add(musicPlaylist);
+
+        monitor.debug(
+            () ->
+                String.format(
+                    "%s: Google Music exporting playlist: %s", jobId, musicPlaylist.getId()));
+
+        // Add playlist id to continuation data
+        continuationData.addContainerResource(new IdOnlyContainerResource(musicPlaylist.getId()));
+      }
+    }
+
+    ResultType resultType = ResultType.CONTINUE;
```

Comment:
per offline chat we should remove this and define it in the conditional (that defines your `nextPageData`) above

Reply:
Updated resultType based on the nextPageData condition.